packages/melos/test/commands/run_test.dart

```diff
@@ -361,5 +361,195 @@ melos run test_script
         ),
       );
     });
+
+    test('throws an error if neither run, steps, nor exec are provided',
+        () async {
+      final workspaceDir = await createTemporaryWorkspace(
+        runPubGet: true,
+        configBuilder: (path) => MelosWorkspaceConfig(
+          path: path,
+          name: 'test_package',
+          packages: [
+            createGlob('packages/**', currentDirectoryPath: path),
+          ],
+          scripts: const Scripts({
+            'test_script': Script(
+              name: 'test_script',
+            ),
+          }),
+        ),
+      );
+
+      await createProject(
+        workspaceDir,
+        const PubSpec(name: 'a'),
+      );
+
+      final logger = TestLogger();
+      final config = await MelosWorkspaceConfig.fromWorkspaceRoot(workspaceDir);
+      final melos = Melos(
+        logger: logger,
+        config: config,
+      );
+
+      expect(() => melos.run(scriptName: 'test_script'), throwsException);
+    });
+
+    test(
+        'throws an error if neither run or steps are provided, and exec '
+        'are options', () async {
+      final workspaceDir = await createTemporaryWorkspace(
+        runPubGet: true,
+        configBuilder: (path) => MelosWorkspaceConfig(
+          path: path,
+          name: 'test_package',
+          packages: [
+            createGlob('packages/**', currentDirectoryPath: path),
+          ],
+          scripts: const Scripts({
+            'test_script': Script(
+              name: 'test_script',
+              exec: ExecOptions(),
+            ),
+          }),
+        ),
+      );
+
+      await createProject(
+        workspaceDir,
+        const PubSpec(name: 'a'),
+      );
+
+      final logger = TestLogger();
+      final config = await MelosWorkspaceConfig.fromWorkspaceRoot(workspaceDir);
+      final melos = Melos(
+        logger: logger,
+        config: config,
+      );
+
+      expect(() => melos.run(scriptName: 'test_script'), throwsException);
+    });
+  });
+
+  group('multiple scripts', () {
+    test(
+        'verifies that a melos script can successfully call another '
+        'script as a step and execute commands', () async {
```

Comment:
Could you add a test where another script that contains `steps` is included in the list of `steps`?

Reply:
I added two tests: one tests the scenario where a script with steps calls another script with steps, and the other test expects an exception related to detecting a recursive call in the script that leads to an infinite loop 17b701a2e692e8e8664b9cfa8f596da0bd5ff0f1.